lib/chef/provider/package/debian.rb

```diff
@@ -0,0 +1,135 @@
+#
+# Author:: Kapil Chouhan (<kapil.chouhan@msystechnologies.com>)
+# Copyright:: Copyright 2008-2018, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/provider/package"
+
+class Chef
+  class Provider
+    class Package
+      class Debian < Chef::Provider::Package
```

Comment:
This should actually be written as a module not a class since the Chef::Provider::Package::Debian class should not be used directly.

That means you'll need to use the included hook:

```
class Chef
  class Provider
    class Package
      module Debian
        def self.included(base)
          base.class_eval do
[... all the normal contents go here...]
          end
        end
      end
    end
  end
end